### PR TITLE
#162170979 User signup functionality

### PR DESCRIPTION
**Pull request by henperi**
#### What does this PR do?
- Integrates functionality for users to be able to fill a form in order to signup on the platform

#### Description of Task to be completed?
- Create signup class component
- Create the UI according to the mockup and make it responsive.
- Create the signup form as a stateless functional component to receive props.

#### How should this be manually tested?
- git clone project
- start server executing `npm run start`
- navigate to the home page and click signup
- fill out the signup form and click submit

#### What are the relevant pivotal tracker stories?
- [#162170979](https://www.pivotaltracker.com/story/show/162170979)

#### Any background context you want to add?
- None

#### Screenshots

### CODE REVIEW

**Review by edogbosunny — APPROVED** [review]

**Review by pitaz — APPROVED** [review]

**Review by akpante3 — APPROVED** [review]

**Review by el-Joft — APPROVED** [review]

**Comment by akpante3** [comment]
hi! @henperi  Nice work, but you can do better on this fewature

**Comment by akpante3** [comment]
hi @henperi  Nice work, but I don't like how you implemented this feature...you can do better.

File: src/component/Layout/MainLayout/MainLayout.jsx [hunk 1]
```diff
@@ -8,16 +9,22 @@ import Footer from '../../ui/footer/Footer';
...
-const MainLayout = ({ children }) => (
-  <div>
-    <Header />
-    {children}
+export const MainLayout = ({ children, auth = {} }) => (
+  <Fragment>
+    <Header isAuthenticated={auth.isAuthenticated} username={auth.user.user.username} />
+    <main role="main" className="pt-5">
+      <div className="container-fluid">{children}</div>
+    </main>
```
Comment (sudhons) [inline src/component/Layout/MainLayout/MainLayout.jsx:17]: @henperi, please why is the `MainLayout` component connected to redux store? Also passing `auth.user.user.username` to the header, will raise an error if the user is not authenticated and why are we having`main`with a padding. Also, what is the purpose of the `container-fluid` div in all of our pages?

**Review by sudhons — CHANGES_REQUESTED** [review]

File: src/component/Layout/MainLayout/MainLayout.jsx (hunk 1 above)
Comment (henperi) [reply src/component/Layout/MainLayout/MainLayout.jsx:17]: I have fixed the main layout connection to redux. Thanks. I have also fixed the ```auth.user.user.username``` issue. 

I previously set the navigation bar to be fixed at the top of the website. So to make sure the nav doesn't sit on the main I had to give it top padding. Outside of this, the main section looks too close to the navigation bar. A little padding adds some aesthetic pleasance in my opinion. 

The ```container-fluid``` or ```container``` class is a basic requirement while using bootstrap 4 grid layout system. 
screen shot 2019-01-28 at 6 20 26 am

Please refer to the bootstrap docs for more info https://getbootstrap.com/docs/4.2/layout/overview/

**Pull request merged**